tools/clang/lib/SPIRV/EmitSPIRVAction.cpp

```diff
@@ -983,35 +980,81 @@ class SPIRVEmitter : public ASTConsumer {
-            theBuilder.createCompositeExtract(returnType, arg0Id, {i});
+            theBuilder.createCompositeExtract(returnTypeId, arg0Id, {i});
         const uint32_t vec1member =
-            theBuilder.createCompositeExtract(returnType, arg1Id, {i});
+            theBuilder.createCompositeExtract(returnTypeId, arg1Id, {i});
         const uint32_t multId = theBuilder.createBinaryOp(
-            multSpvOp, returnType, vec0member, vec1member);
+            multSpvOp, returnTypeId, vec0member, vec1member);
         multIds.push_back(multId);
       }
       // Add all the multiplications.
       result = multIds[0];
       for (unsigned int i = 1; i < vec0Size; ++i) {
-        const uint32_t additionId =
-            theBuilder.createBinaryOp(addSpvOp, returnType, result, multIds[i]);
+        const uint32_t additionId = theBuilder.createBinaryOp(
+            addSpvOp, returnTypeId, result, multIds[i]);
         result = additionId;
       }
       return result;
     }
   }
 
-  /// Processes the given expr, casts the result into the given bool (vector)
-  /// type and returns the <result-id> of the casted value.
-  uint32_t castToBool(const Expr *expr, QualType toBoolType) {
-    // Converting to bool means comparing with value zero.
+  uint32_t processIntrinsicAny(const CallExpr *callExpr) {
+    const uint32_t returnType =
+        typeTranslator.translateType(callExpr->getType());
 
-    const spv::Op spvOp = translateOp(BO_NE, expr->getType());
-    const uint32_t boolType = typeTranslator.translateType(toBoolType);
-    const uint32_t originalVal = doExpr(expr);
-    const uint32_t zeroVal = getValueZero(expr->getType());
+    // 'any' functoin takes only 1 parameter.
+    assert(callExpr->getNumArgs() == 1u);
+    const Expr *arg = callExpr->getArg(0);
+    const QualType argType = arg->getType();
+    bool isScalar = argType->isBooleanType() || argType->isIntegerType() ||
+                    argType->isFloatingType();
+    bool isVec1 = hlsl::IsHLSLVecType(argType) &&
+                  hlsl::GetHLSLVecSize(argType) == 1u &&
+                  (hlsl::GetHLSLVecElementType(argType)->isBooleanType() ||
+                   hlsl::GetHLSLVecElementType(argType)->isIntegerType() ||
+                   hlsl::GetHLSLVecElementType(argType)->isFloatingType());
+
+    if (isScalar || isVec1) {
+      // For a scalar or vector of 1 scalar, we can simply cast to boolean.
+      return castToBool(arg, callExpr->getType());
+    }
+    if (hlsl::IsHLSLVecType(argType)) {
+      // First cast the vector to a vector of booleans, then use OpAny
+      uint32_t boolVecId = castToBool(arg, callExpr->getType());
+      return theBuilder.createUnaryOp(spv::Op::OpAny, returnType, boolVecId);
+    }
+
+    emitError("Intrinsic function 'any' doesn't support matrix arguments yet.");
+    return 0;
+  }
 
-    return theBuilder.createBinaryOp(spvOp, boolType, originalVal, zeroVal);
+  uint32_t processIntrinsicAll(const CallExpr *callExpr) {
```

Comment:
This method has largely the same structure as the one for `any`. Maybe we can unify them into one by supplying an additional parameter for `spv::Op`?